backend/src/packages/stripe/stripe.repository.ts

```diff
@@ -0,0 +1,177 @@
+        on_behalf_of: businessStripeId,
+        transfer_data: {
+          destination: businessStripeId,
+        },
+        metadata,
+      },
+      mode: CHECKOUT_MODE,
+      success_url: constructUrl(
+        this.config.APP.FRONTEND_BASE_URL,
+        FrontendPath.PAYMENTS,
+        { [StripeOperationStatus.SUCCESS]: true },
+      ),
+      cancel_url: constructUrl(
+        this.config.APP.FRONTEND_BASE_URL,
+        FrontendPath.PAYMENTS,
+        { [StripeOperationStatus.CANCEL]: true },
+      ),
+    });
+
+    return session.url;
+  }
+
+  public createExpressAccount(
+    user: UserEntityObjectWithGroupAndBusinessT,
+  ): Promise<Stripe.Account> {
+    const accountParameters: Stripe.AccountCreateParams = {
+      type: CONNECT_ACCOUNT_TYPE,
+      email: user.email,
+      default_currency: DEFAULT_CURRENCTY,
+      business_profile: {
+        // TODO: this is temporary, I guess, we would eventually have a personal business page in our app
+        // Anyway, this field will be shown to the enduser during registration unless we prefill it prematurely
+        url:
+          process.env.NODE_ENV === 'production'
+            ? this.config.APP.FRONTEND_BASE_URL
+            : constructUrl(
+                this.config.APP.FRONTEND_BASE_URL,
+                `/business/${user.business.id}`,
+              ),
+        mcc: STRIPE_TOWING_SERVICES_MCC,
+        name: user.business.companyName,
+      },
+      business_type: CONNECT_ACCOUNT_BUSINESS_TYPE,
+      individual: {
+        first_name: user.firstName,
+        last_name: user.lastName,
+        email: user.email,
+      },
+      metadata: {
+        userId: user.id,
+        businessId: user.business.id,
+      },
+    };
+
+    return this.stripe.accounts.create(accountParameters);
+  }
+
+  public retrieveExpressAccount(id: string): Promise<Stripe.Account> {
+    return this.stripe.accounts.retrieve(id);
+  }
+
+  public generateExpressAccountOnboardingLink(
+    account: Stripe.Account,
+  ): Promise<Stripe.AccountLink> {
+    return this.stripe.accountLinks.create({
+      account: account.id,
+      refresh_url: `${this.config.APP.FRONTEND_BASE_URL}${FrontendPath.SETUP_PAYMENT}?refresh=true`,
+      return_url: `${this.config.APP.FRONTEND_BASE_URL}${FrontendPath.SETUP_PAYMENT}?success=true`,
+      type: ONBOARDING_LINK_TYPE,
+    });
+  }
+
+  public collectPaymentIntents(
+    options: GetPaymentsRequest,
+  ): Promise<PaymentIntentWithMetadata[]> {
+    return this.stripe.paymentIntents
+      .search(buildPaymetnsRequestQuery(options))
+      .autoPagingToArray({ limit: 1000 }) as Promise<
+      PaymentIntentWithMetadata[]
+    >;
+  }
+
+  private calculateTotal({
+    price,
+    quantity,
+  }: {
+    price: number;
+    quantity: number;
+  }): number {
+    return price * quantity;
+  }
+
+  private calculateApplicationFee(amount: number): number {
+    return amount * this.config.BUSINESS.APPLICATION_FEE_AMOUNT;
+  }
+}
+
+export { StripeRepository };
```

Comment:
It seems we don't need repository for stripe as we dont use it as storage, part of the logic can be in service, part in helpers